### ISSUE TITLE
Messages getting stuck in flight in SQS

### ISSUE THREAD

**Issue by 3pdev-alex** [localstack/localstack#3113]
# Type of request: This is a ...
[X] bug report
[ ] feature request

# Detailed description
Using PHP, I put a message into an SQS queue in localstack. I then use a different PHP service to pull the message from the queue and act on it. The service is running fine and doing what it should, but oftentimes (like, a _lot_) after inserting the message into the queue, get-queue-attributes is reporting 0 ApproximateNumberOfMessages and ApproximateNumberOfMessagesDelayed, and several ApproximateNumberOfMessagesNotVisible.

## Expected behavior
When I put a message into a SQS queue, it should be available for external consumption and processing.

## Actual behavior
Messages are getting stuck in flight and are unavailable for consumption.

# Steps to reproduce
Put a message into a queue, then check the queue attributes using `awslocal sqs get-queue-attributes`

## Command used to start LocalStack
I'm using the Docker setup with a simple docker-entrypoint-initaws.d that includes `awslocal sqs create-queue --queue-name my-queue-for-development` several times for multiple queues.

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
```
    public function receiveMessage($maxNumberOfMessages=1, $waitTimeSeconds=2){
        try {
            $result = $this->client->receiveMessage([
                'QueueUrl' => 'my_local_queue_url',
                'MaxNumberOfMessages' => $maxNumberOfMessages,
                'WaitTimeSeconds' => $waitTimeSeconds,
                'AttributeNames' => ['ApproximateReceiveCount']
            ]);
            $this->errorMessage = "";
        } catch (\Guzzle\Service\Exception\ValidationException $ve) {
            $this->errorMessage = "swSqsClient: ".$ve->getMessage();
        } catch (\Aws\Sqs\Exception\SqsException $e) {
            $this->errorMessage = "swSqsClient: ".$e->getMessage();
        }
        return $result['messages'];
    }
```

$this->client being an instance of Aws\Sqs\SqsClient in PHP.

┆Issue is synchronized with this [Jira Bug](https://localstack.atlassian.net/browse/LOC-291) by [Unito](https://www.unito.io/learn-more)

**Comment by whummer** [comment]
Thanks for reporting @3pdev-alex . Can you please share a few more details about your environment? We have recently pushed a couple of important performance fixes - can you please do a `docker pull localstack/localstack` and confirm whether the issue still persists with the `latest` version? Thanks for your help.

**Comment by 3pdev-alex** [comment]
Sorry - I got busy and haven't had time to come back to this. I'll try updating this weekend.

I have noticed that problem seems to get more prevalent the more queues I'm using. If I've only got services pushing and pulling to and from 1 queue this almost never happens. However, if I'm running my full setup as it does in production it interacts with a minimum of 5 queues - it's at this point the issue starts to happen with regularity.
